crates/sui-framework/sources/balance.move

```diff
@@ -63,26 +63,46 @@ module sui::balance {
         Balance { value: 0 }
     }
 
+    spec zero {
+        aborts_if false;
```

Comment:
I'm not sure if we want to have any specs here. I'm hoping to show that `sui move prove` works even with specs present. But we can maybe show this in a different `example_specs` move package?